cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/provisioner/TetheringConf.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.tethering.runtime.spi.provisioner;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.PrivateModule;
+import com.google.inject.Scopes;
+import io.cdap.cdap.api.metrics.MetricsCollectionService;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.guice.ConfigModule;
+import io.cdap.cdap.data.runtime.StorageModule;
+import io.cdap.cdap.data.runtime.SystemDatasetRuntimeModule;
+import io.cdap.cdap.data.runtime.TransactionExecutorModule;
+import io.cdap.cdap.internal.tethering.NamespaceAllocation;
+import io.cdap.cdap.internal.tethering.PeerInfo;
+import io.cdap.cdap.internal.tethering.TetheringStatus;
+import io.cdap.cdap.internal.tethering.TetheringStore;
+import io.cdap.cdap.metrics.collect.LocalMetricsCollectionService;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import org.apache.tephra.runtime.TransactionModules;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Configuration for the Tethering Provisioner.
+ */
+public class TetheringConf {
+  public static final String TETHERED_INSTANCE_PROPERTY = "tetheredInstanceName";
+  public static final String TETHERED_NAMESPACE_PROPERTY = "tetheredNamespace";
+
+  private final String tetheredInstanceName;
+  private final String tetheredNamespace;
+
+  private TetheringConf(String tetheredInstanceName, String tetheredNamespace) {
+    this.tetheredInstanceName = tetheredInstanceName;
+    this.tetheredNamespace = tetheredNamespace;
+  }
+
+  /**
+   * Create the conf from a property map while also performing validation that tethered connection exists.
+   */
+  public static TetheringConf fromProperties(Map<String, String> properties) {
+    String tetheredInstanceName = getString(properties, TETHERED_INSTANCE_PROPERTY);
+    String tetheredNamespace = getString(properties, TETHERED_NAMESPACE_PROPERTY);
+
+    Injector injector = createInjector();
```

Comment:
Is it possible to move this someplace so that only one injector is created? and then inject store info?